test/session_test/password_test.rb

```diff
@@ -86,9 +86,21 @@ def test_credentials
         assert_equal({ password: "<protected>", login: "login" }, session.credentials)
       end
 
+      def test_non_permitted_action_controller_parameters_credentials
```

Comment:
If you don't force a `to_h` and expect a hash-like object instead you'll need to re-add the development dependency on actionpack and specifically test that `AC::Params` works properly.